### PR TITLE
Feat/aapd 1466 application reference

### PR DESCRIPTION
**Pull request by topickering**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-1466

## Description of change

Adds the 'your application' section to the HAS Appeal journey, with 4 questions.

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- X My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/date/question.js [hunk 1]
```diff
@@ -19,16 +19,18 @@ class DateQuestion extends Question {
...
+	constructor({ title, question, fieldName, validators, hint, url }) {
 		super({
 			title: title,
 			viewFolder: 'date',
 			fieldName: fieldName,
 			question: question,
 			validators: validators,
-			hint: hint
+			hint: hint,
+			url: url
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/dynamic-components/date/question.js:33]:
```suggestion
			hint,
			url
```

File: packages/forms-web-app/src/dynamic-forms/has-appeal-form/journey.js [hunk 2]
```diff
@@ -89,7 +89,12 @@ class HasAppealFormJourney extends Journey {
 			new Section('Application', 'application')
 				.addQuestion(questions.uploadOriginalApplicationForm)
 				.addQuestion(questions.uploadChangeOfDescriptionEvidence)
-				.addQuestion(questions.uploadApplicationDecisionLetter)
+				.addQuestion(questions.uploadApplicationDecisionLetter),
+			new Section('Your application', 'your-application')
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/has-appeal-form/journey.js:93]: Nooooooo not an inevitable conflict 
reeee

**Review by Si3rr4wow — DISMISSED** [review]

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/date/question.js (hunk 1 above)
Comment (topickering) [reply packages/forms-web-app/src/dynamic-forms/dynamic-components/date/question.js]: done

File: packages/forms-web-app/src/dynamic-forms/has-appeal-form/journey.js (hunk 2 above)
Comment (topickering) [reply packages/forms-web-app/src/dynamic-forms/has-appeal-form/journey.js]: Oops, I've put the section in the right place following a rebase

**Review by Si3rr4wow — APPROVED** [review]

**Pull request merged**